src/asami/core.cljc

```diff
@@ -274,27 +276,31 @@
   :db-after     database value after the transaction
   :tx-data      sequence of datoms produced by the transaction
   :tempids      mapping of the temporary IDs in entities to the allocated nodes"
-  [{:keys [name state] :as connection} :- ConnectionType
-   {:keys [tx-data tx-triples] :as tx-info} :- TransactData]
-  (let [op (fn []
-             (let [tx-id (count (:history @state))
-                   as-datom (fn [assert? [e a v]] (->Datom e a v tx-id assert?))
-                   {:keys [graph history] :as db-before} (:db @state)
-                   [triples removals tempids] (if tx-triples       ;; is this inserting raw triples?
-                                                [tx-triples nil {}]
-                                                ;; capture the old usage which didn't have an arg map
-                                                (build-triples db-before (or tx-data tx-info)))
-                   [db-before db-after] (storage/transact-data connection triples removals)]
-               {:db-before db-before
-                :db-after db-after
-                :tx-data (concat
-                          (map (partial as-datom false) removals)
-                          (map (partial as-datom true) triples))
-                :tempids tempids}))]
-    #?(:clj (future (op))
-       :cljs (let [d (delay (op))]
-               (force d)
-               d))))
+  ([connection tx-info]
+   (transact connection tx-info nil))
+  ([{:keys [name state] :as connection} :- ConnectionType
+    {:keys [tx-data tx-triples] :as tx-info} :- TransactData
+    {:keys [executor] :as options}]
```

Comment:
I don't know about adding a whole new argument here. It looks to me like the new `tx-info` argument that Datomic introduced seemed to have made this a map specifically to be able to provide extra arguments like this one.

The only problem with adding it as a new key for `tx-info` is if it contains a seq of transaction data instead of being a map that contains a `:tx-data` key. But that option has just been provided for backward compatibility with the previous API for Datomic. So I don't think it's something to worry about.

So I'm suggesting that `TransactData` (line 243) is expanded to:
```clojure
(def TransactData (s/if map?                                       
                    {(s/optional-key :tx-data) [s/Any]
                     (s/optional-key :tx-triples) [[(s/one s/Any "entity")
                                                    (s/one s/Any "attribute")
                                                    (s/one s/Any "value")]]
                     (s/optional-key :executor) s/Any}
                    [s/Any]))
```
and then line 282 gets updated to:
``` clojure
    {:keys [tx-data tx-triples executor] :as tx-info} :- TransactData
```

Reply:
Sounds good to me. Thanks for the review.

Reply:
amended